pkg/template/event_test.go

```diff
@@ -357,73 +358,107 @@ func TestResolveParams(t *testing.T) {
 		bindingParams: []triggersv1.Param{
-			bldr.Param("p1", "val1"),
-			bldr.Param("p2", "$(body.p2)"),
-		},
-		template: bldr.TriggerTemplate("tt-name", ns,
-			bldr.TriggerTemplateSpec(
-				bldr.TriggerTemplateParam("p2", "", "defaultVal"),
-			),
-		),
-		want: []triggersv1.Param{
-			bldr.Param("p1", "val1"),
-			bldr.Param("p2", "defaultVal"),
+			{Name: "p1", Value: "val1"},
+			{Name: "p2", Value: "$(body.p2)"},
 		},
-	}, {
-		name: "default values do not override event values",
-		bindingParams: []triggersv1.Param{
-			bldr.Param("p1", "val1"),
+		template: &triggersv1.TriggerTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tt-name",
+				Namespace: ns,
+			},
+			Spec: triggersv1.TriggerTemplateSpec{
+				Params: []triggersv1.ParamSpec{{
+					Name:        "p2",
+					Description: "",
+					Default:     ptr.String("defaultVal"),
+				}},
+			},
 		},
-		template: bldr.TriggerTemplate("tt-name", ns,
-			bldr.TriggerTemplateSpec(
-				bldr.TriggerTemplateParam("p1", "", "defaultVal"),
-			),
-		),
 		want: []triggersv1.Param{
-			bldr.Param("p1", "val1"),
+			{Name: "p2", Value: "defaultVal"},
+			{Name: "p1", Value: "val1"},
 		},
+	}, {
+		name:          "default values do not override event values",
+		bindingParams: []triggersv1.Param{{Name: "p1", Value: "val1"}},
+		template: &triggersv1.TriggerTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tt-name",
+				Namespace: ns,
+			},
+			Spec: triggersv1.TriggerTemplateSpec{
+				Params: []triggersv1.ParamSpec{{
+					Name:        "p1",
+					Description: "",
+					Default:     ptr.String("defaultVal"),
+				}},
+			},
+		},
+		want: []triggersv1.Param{{Name: "p1", Value: "val1"}},
 	}, {
 		name: "combination of static values and JSONPath expressions",
 		body: json.RawMessage(`{"foo": "fooValue", "bar": "barValue"}`),
 		bindingParams: []triggersv1.Param{
-			bldr.Param("p1", "Event values are - foo: $(body.foo); bar: $(body.bar)"),
+			{Name: "p1", Value: "Event values are - foo: $(body.foo); bar: $(body.bar)"},
+		},
+		template: &triggersv1.TriggerTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tt-name",
+				Namespace: ns,
+			},
+			Spec: triggersv1.TriggerTemplateSpec{},
 		},
-		template: bldr.TriggerTemplate("tt", ns),
 		want: []triggersv1.Param{
-			bldr.Param("p1", "Event values are - foo: fooValue; bar: barValue"),
+			{Name: "p1", Value: "Event values are - foo: fooValue; bar: barValue"},
 		},
 	}, {
 		name: "values with newlines",
 		body: json.RawMessage(`{"foo": "bar\r\nbaz"}`),
-		template: bldr.TriggerTemplate("tt-name", "",
-			bldr.TriggerTemplateSpec(
-				bldr.TriggerTemplateParam("param1", "", ""),
-				bldr.TriggerTemplateParam("param2", "", ""),
-			),
-		),
+		template: &triggersv1.TriggerTemplate{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "tt-name",
+				Namespace: ns,
+			},
+			Spec: triggersv1.TriggerTemplateSpec{
+				Params: []triggersv1.ParamSpec{{
+					Name:        "param1",
+					Description: "",
```

Comment:
nit: you can remove fields that just contain the empty value.

Reply:
done.